### PR TITLE
Add `TextArea` as a private component

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description
- Add `TextArea` as a private model (as it has been requested several times now as a custom component)
- Add unit tests for currently private models `UserInput` and `TextArea`
- Remove  "input_type" from `UserInput` for now
- Consolidate CSS between `TextArea` and `UserInput`
- Update `box-shadow `variables based on latest Figma design

Not related but squeezed this in:
- Tidy some docs (remove `Page.path` when not necessary) - In general we shouldn't add an `id` or `path` if it's not required in the config for the examples

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/examples/_dev/app.py
Comment (huong-li-nguyen) [inline vizro-core/examples/_dev/app.py:13]: Not related to this PR, but we've already mentioned it, but worth discussing whether we even need a `Form` model or if we just want to fall back on the `Container` and add a default action if there is a Button inside. Will be relevant for the model interactions that are yet to come @maxschulz-COL

File: vizro-core/src/vizro/models/_components/form/_user_input.py
Comment (antonymilne) [inline vizro-core/src/vizro/models/_components/form/_user_input.py:47]: Wouldn't the same be true for `"number"` also?

Actually I think we may well want to just fix `input_type="text"` here and not expose the field at all. Doesn't really matter for now since it's not public, but if you agree then let's just remove the field entirely.

File: vizro-core/src/vizro/models/_components/form/_text_area.py
Comment (antonymilne) [inline vizro-core/src/vizro/models/_components/form/_text_area.py:27]:
```suggestion
    # TODO: before making public consider naming this field (or giving an alias) label instead of title
    title: str = Field("", description="Title to be displayed")
```

File: vizro-core/src/vizro/models/_components/form/_user_input.py
Comment (antonymilne) [inline vizro-core/src/vizro/models/_components/form/_user_input.py:29]:
```suggestion
    # TODO: before making public consider naming this field (or giving an alias) label instead of title
    title: str = Field("", description="Title to be displayed")
```

File: vizro-core/src/vizro/models/_components/form/_text_area.py
Comment (antonymilne) [inline vizro-core/src/vizro/models/_components/form/_text_area.py:38]: I guess we would make this `dbc.Label` in future if we're going all in on bootstrap. Judging by the docs it would make a very small difference in practice compared to `html.Label` though.

**Review by antonymilne — APPROVED** [review]
Looks good! 

FYI for when it does come to working out the form components properly, we should review https://wtforms.readthedocs.io/en/3.1.x/ and https://github.com/pydantic/FastUI/blob/main/demo/forms.py.

File: vizro-core/src/vizro/models/_components/form/_user_input.py
Comment (huong-li-nguyen) [reply vizro-core/src/vizro/models/_components/form/_user_input.py:47]: Yes, you are right! I think the only input types that are quite useful are `email` and `password`. This will most likely be used for authentication purposes though, where I think we currently recommend leveraging existing systems. 

So let me remove it for now and see how people might want to use it 👍

File: vizro-core/src/vizro/models/_components/form/_text_area.py
Comment (huong-li-nguyen) [reply vizro-core/src/vizro/models/_components/form/_text_area.py:38]: Yes, agree 👍  Would probably leave it for now, so it's consistent with the other components and then we change everything in one go.

File: vizro-core/examples/_dev/app.py
Comment (petar-qb) [inline vizro-core/examples/_dev/app.py:9]: Do we need to undo the changes for this file before merging on the main? 

It seems that we've changed the file content in [this PR](https://github.com/mckinsey/vizro/pull/277). I'm wondering what is the final content for this example that we should keep.

File: vizro-core/src/vizro/models/_components/form/_text_area.py
Comment (petar-qb) [inline vizro-core/src/vizro/models/_components/form/_text_area.py:33]: This way of configuring TextArea and UserInput: 
- `_set_actions = _action_validator_factory("value")`
- `debouncing=True`

causes that if the "actions" property is defined, actions chain will be triggered when the enter key is pressed or `when the component loses focus`.

If we want not to trigger actions chain every time when the component loses focus, but only if the enter key is pressed, then we should set `_action_validator_factory` argument to `n_submit`.

**Review by petar-qb — APPROVED** [review]
Looks really good!

**Pull request merged**
